src/shim.ts

```diff
@@ -6,43 +30,293 @@ import {
     Orchestrator,
 } from "./classes";
 import { DurableEntityBindingInfo } from "./durableentitybindinginfo";
-import { DurableOrchestrationBindingInfo } from "./durableorchestrationbindinginfo";
 import { OrchestratorState } from "./orchestratorstate";
+import { DurableOrchestrationInput } from "./testingUtils";
 
 /**
  * Enables a generator function to act as an orchestrator function.
  *
- * Orchestration context methods can be access
  * @param fn the generator function that should act as an orchestrator
- * @example Initialize an orchestrator
+ * @example Register an orchestrator using `app.generic()`
  * ```javascript
+ * const { app } = require("@azure/functions")
  * const df = require("durable-functions");
  *
- * module.exports = df.orchestrator(function*(context) {
- *     // function body
- * });
+ * app.generic('DurableFunctionsOrchestratorJS', {

```

Comment:
I've simplified the examples a lot, so that they're not same complexity or less than what we had earlier. Lmk if you think this is better or if we should get rid of the examples entirely

Reply:
Intellisense is a priority for us in the new model, but not a blocker for this PR. I will spend time looking at the docs/examples at a later time